src/object.rs

```diff
@@ -859,3 +878,17 @@ impl<T: IsA<Object> + ?Sized> Clone for WeakRef<T> {
 
 unsafe impl<T: IsA<Object> + Send + Sync + ?Sized> Send for WeakRef<T> {}
 unsafe impl<T: IsA<Object> + Send + Sync + ?Sized> Sync for WeakRef<T> {}
+
+pub trait Classed<C>
+  where C: Wrapper {
+    fn get_class(&self) -> &C::ClassType;
+}
```

Comment:
New trait that only exists so we can implement it generically below for all GObjects